### ISSUE TITLE
Incorrect assumption that bot user ID is the same as the application client ID

### ISSUE THREAD

**Issue by dxsmiley** [interactions-py/interactions.py#112]
I was trying to get started with the library using the example code [here](https://discord-py-slash-command.readthedocs.io/en/latest/gettingstarted.html), but run into this error:

```
Task exception was never retrieved
future: <Task finished coro=<SlashCommand.sync_all_commands() done, defined at .../discord_slash/client.py:293> exception=NotFound('404 Not Found (error code: 10002): Unknown Application',)>
Traceback (most recent call last):
  File ".../discord_slash/client.py", line 308, in sync_all_commands
    await self.req.put_slash_commands(slash_commands=cmds["global"], guild_id=None)
  File ".../discord/http.py", line 243, in request
    raise NotFound(r, data)
discord.errors.NotFound: 404 Not Found (error code: 10002): Unknown Application
```

I traced it down to this bit of the code, specifically line 75. 

https://github.com/eunwoo1104/discord-py-slash-command/blob/2a9ebcfe1ce5ca376e13ab038810a9eebf2a414f/discord_slash/http.py#L66-L79

The endpoint requires the [application id](https://discord.com/developers/docs/interactions/slash-commands#create-global-application-command), which is _not guaranteed to be the same_ as the bot's user ID.

I suspect that this is not the only point in the codebase with this issue.

**Comment by Sntx626** [comment]
As it seems right now it doesn't appear to be easy to account for that and therefore won't be fixed. Compare with this Issue: https://github.com/eunwoo1104/discord-py-slash-command/issues/108

**Comment by dxsmiley** [comment]
That's disappointing :/

**Comment by eunwoo1104** [comment]
Late comment but anyways fixed at #113.
